src/domain/relay/limit-addresses.mapper.spec.ts

```diff
@@ -0,0 +1,722 @@
+        getAddress(faker.finance.ethereumAddress()),
+      ];
+      const singleton = getSafeL2SingletonDeployment({
+        version,
+        network: chainId,
+      })!.networkAddresses[chainId];
+      const data = createProxyWithNonceEncoder()
+        .with('singleton', getAddress(singleton))
+        .with('initializer', setupEncoder().with('owners', owners).encode())
+        .encode();
+      // ProxyFactory address (singletons are checked for official mastercopies so we need not check this)
+      const to = getAddress(faker.finance.ethereumAddress());
+
+      const expectedLimitAddresses = await target.getLimitAddresses({
+        chainId,
+        data,
+        to,
+      });
+      expect(expectedLimitAddresses).toStrictEqual(owners);
+    });
+
+    it('should throw when creating an unofficial Safe', async () => {
+      const chainId = faker.string.numeric();
+      const owners = [
+        getAddress(faker.finance.ethereumAddress()),
+        getAddress(faker.finance.ethereumAddress()),
+      ];
+      // Unofficial singleton
+      const singleton = getAddress(faker.finance.ethereumAddress());
+      const data = createProxyWithNonceEncoder()
+        .with('singleton', getAddress(singleton))
+        .with('initializer', setupEncoder().with('owners', owners).encode())
+        .encode();
+      // ProxyFactory address (singletons are checked for official mastercopies so we need not check this)
+      const to = getAddress(faker.finance.ethereumAddress());
+
+      await expect(
+        target.getLimitAddresses({
+          chainId,
+          data,
+          to,
+        }),
+      ).rejects.toThrow('Cannot get limit addresses – Invalid transfer');
+    });
+  });
+
+  describe('Validation', () => {
+    it('should throw if not an execTransaction, multiSend or createProxyWithNonceCall', async () => {
+      const chainId = faker.string.numeric();
+      const safe = safeBuilder().build();
+      const safeAddress = getAddress(safe.address);
+      const data = erc20TransferEncoder().encode();
+      // Official mastercopy
+      mockSafeRepository.getSafe.mockResolvedValue(safe);
+
+      await expect(
+        target.getLimitAddresses({
+          chainId,
+          data,
+          to: safeAddress,
+        }),
+      ).rejects.toThrow('Cannot get limit addresses – Invalid transfer');
+    });
+
+    it('should throw if the to address is not valid', async () => {
+      const chainId = faker.string.numeric();
+      const to = '0x000000000000000000000000000000000INVALID';
+      const data = erc20TransferEncoder().encode();
+
+      await expect(
+        target.getLimitAddresses({
+          chainId,
+          data,
+          to,
+        }),
+      ).rejects.toThrow('Invalid recipient or calldata provided');
+    });
+
+    it('should throw if the to address is not hexadecimal', async () => {
+      const chainId = faker.string.numeric();
+      const to = 'not hexidecimal';
```

Comment:
```suggestion
      const to = 'not hexadecimal';
```

Reply:
Fixed in ca9b11ca52cb5aabda07baca59bb1391fb567ec6.